src/index.js

```diff
@@ -120,16 +120,23 @@ class ApiClientError extends Error {
   }
 }
 
-ClientApi.prototype.request = async (url, data, format) => {
+ClientApi.prototype.request = async (url, data, format, method = 'GET') => {
   try {
     let requestConfig = structuredClone(defaultAxiosConfig)
+    requestConfig.method = method
+    requestConfig.url = url
     if (data) {
-      requestConfig.params = { ...requestConfig.params, ...data }
+      if (method === 'GET') {
+        requestConfig.params = data
```

Comment:
Could there be existing params in `requestConfig.params`?

Reply:
The `defaultAxiosConfig` does not have any (and I don't expect we are going to add any extra query parameters).